semeio/reporter/reporter.py

```diff
@@ -0,0 +1,32 @@
+class Reporter:
+    def __init__(self, name):
+        self.name = name
+
+    def _get_suffix_for_value(self, value):
+
+        if isinstance(value, (str, float, int)):
+            return "txt"
+
+        valid_types = ",".join(["string", "float", "int"])
+
+        raise TypeError(
+            "The type of value <{}> must be one of: {} was {}".format(
+                value, valid_types, type(value)
+            )
+        )
+
+    def report(self, key=None, value=None, write=False):
```

Comment:
I think deducing data types will get really hairy. I also don't think it should be on the caller to toggle a `write`-flag.

I suggest that in the first iteration we support:
```python
Reporter.report(self, value: str)
Reporter.publish(self, key: str, value: json)
```

Then we immediately wrap the Python logger (with log-level `info`) around `Reporter.report` and let all `report` calls be appended in a file on disk with name `<reporter_namespace>/report`.

Second, we use the `publish` method to persist structured data as json. We can later extend with `publish_matrix` etc, but I suggest that the default structured data format is json. Each of these calls will in the first iteration induce a dump of the value to `<reporter_namespace>published/<key>.json`.

We need to find a reasonable base-folder for the data dumping. One option is to inject `ENSPATH` into the job description files and use `<enspath>/reporter` as base. This is at least a location the user is expecting output to.

Reply:
We can extend upon this later, but this will be sufficient for our needs now 👍

Reply:
> I think deducing data types will get really hairy. 

I agree, I also thought this was a poor decision.

> I also don't think it should be on the caller to toggle a write-flag.

Unsure.. As long as everything is written to disk as suggested, then of course it doesn't matter. But if we get in a scenario where it would be beneficial to only store some data _somewhere_, then I think it should be up to the caller to decide if the option is available.

Just note that if we decide to go with write-to-disk as default, we would have to keep that as default unless we want to break the API..

Reply:
> Unsure.. As long as everything is written to disk as suggested, then of course it doesn't matter. But if we get in a scenario where it would be beneficial to only store some data somewhere, then I think it should be up to the caller to decide if the option is available.

But the point is that the job does not know whether the data is important or not. Because it is dependent on the setting for which it runs in. Hence, the job should only serve the data and the option of keeping it or not should be configured further up in the call stack (preferably to configuration file).

Reply:
> Just note that if we decide to go with write-to-disk as default, we would have to keep that as default unless we want to break the API..

That is an experimental feature that we will break as soon as we can. When this data goes into the database we will immediately stop dumping then to files...